### PR TITLE
Allow admins to kick members from a team

### PR DESCRIPTION
**Pull request by eluric**
This is an important ability so that team members that are causing grievances can be removed from a team that they may otherwise be unwilling to leave.

EDIT:
Upon abstracting the code for removing players from a team (since the code for kicking a player is largely the same as a player leaving by themself), we encountered CommandInvokeErrors. 
Specifically
`discord.app_commands.errors.CommandInvokeError: Command 'kick_member' raised an exception: NotFound: 404 Not Found (error code: 10008): Unknown Message`

This happens when the code attempts to respond to an interaction in a channel that has been deleted. This did not occur before the abstraction of code because the interaction received a response before the channels were deleted. However, post-abstraction the folllowup response is only sent after the deletion of channels. This is a little hard to get around. We can possibly leave the actual deletion of the channels to the individual cog commands but that is not an ideal solution. We can also pass in the interaction to the context command and allow it to respond to the interaction. 

However, we've chosen to simply catch the error when it happens and ignore it.

### CODE REVIEW

File: cogs/admin.py [hunk 1]
```diff
@@ -172,6 +174,64 @@ async def set_hint_channel(self, interaction: discord.Interaction):
...
+        text_channel = guild.get_channel(int(team.text_channel_id))
+        voice_channel = guild.get_channel(int(team.voice_channel_id))
+
+        # delete roles and channels
+        await text_channel.delete()
+        await voice_channel.delete()
+        await category_channel.delete()
+        await role.delete()
+
+        await remove_team(team_name)
```
Comment (SkellyBG) [inline cogs/admin.py:228]: can we DRY this and leave_team?

Comment (eluric) [reply cogs/admin.py:228]: What's DRY?

Comment (eluric) [reply cogs/admin.py:228]: I looked it up and I'll try to

Comment (SkellyBG) [reply cogs/admin.py:228]: tyty ^^ DRY is Don't Repeat Yourself

**Comment by eluric** [comment]
Since leave_team and kick_member had essentially the same code, it has been abstracted out of both methods and placed in a separate file.
Typecasting has also been removed to fit with Pydantic implemented in #15

**Comment by SkellyBG** [comment]
hey nice work! I feel like nitpicking ;;-; but I think this is too DRY. Generally when you DRY up a piece of code, you want its to be a logical unit that's testable - which this context code isn't quite there. A better option would be to say, have a function that handles the deletion of a player, and another function that handles the deletion of a team + all the channels - then they would be much more testable. If that doesn't quite make sense, let me know!

**Comment by SkellyBG** [comment]
mhmm I think, at least for me, that cogs are the "front-end" of discord bots and should handle the messages instead of context - passing in a boolean to control how the message is sent feels like a code smell to me

**Comment by eluric** [comment]
> mhmm I think, at least for me, that cogs are the "front-end" of discord bots and should handle the messages instead of context - passing in a boolean to control how the message is sent feels like a code smell to me

I'm not sure what you mean by code smell but I've fixed it.

File: src/context/team.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+from src.models.player import Player
+
+
+async def get_team_channels(guild: discord.Guild, team_name: str):
+    team = await get_team(team_name)
+
+    voice_channel = guild.get_channel(int(team.voice_channel_id))
+    text_channel = guild.get_channel(int(team.text_channel_id))
+    category_channel = guild.get_channel(int(team.category_channel_id))
```
Comment (SkellyBG) [inline src/context/team.py:16]: don't need casting here

File: src/context/team.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+    role = await remove_role_and_player(guild, player, member)
+
+    if not role:
+        return None
+
+    team_name = player.team_name
+
+    channels = await get_team_channels(guild, team_name)
+
```
Comment (SkellyBG) [inline src/context/team.py:63]: we can move this to after the team_member check

**Review by SkellyBG — APPROVED** [review]
LGTM - just some small nitpicks!

File: src/context/team.py (hunk 2 above)
Comment (eluric) [reply src/context/team.py:16]: whoops forgot to change that after testing

**Pull request merged**